pkg/wallet/internal/mapping/mapping_list_actions.go

```
@@ -0,0 +1,187 @@
+		if hash, err := chainhash.NewHashFromHex(action.TxID); err == nil {
+			txidHash = *hash
+		}
+	}
+
+	if action.Satoshis < 0 {
+		panic("negative satoshis value encountered: data integrity issue")
+	}
+
+	result := sdk.Action{
+		Txid:        txidHash,
+		Satoshis:    uint64(action.Satoshis),
+		Status:      mapActionStatus(action.Status),
+		IsOutgoing:  action.IsOutgoing,
+		Description: action.Description,
+		Labels:      action.Labels,
+		Version:     action.Version,
+		LockTime:    action.LockTime,
+		Inputs:      slices.Map(action.Inputs, mapActionInput),
+		Outputs:     slices.Map(action.Outputs, mapActionOutput),
+	}
+
+	return result
+}
+
+// mapActionStatus maps string status to sdk.ActionStatus
+func mapActionStatus(status string) sdk.ActionStatus {
+	switch status {
+	case "completed":
+		return sdk.ActionStatusCompleted
+	case "unprocessed":
+		return sdk.ActionStatusUnprocessed
+	case "sending":
+		return sdk.ActionStatusSending
+	case "unproven":
+		return sdk.ActionStatusUnproven
+	case "unsigned":
+		return sdk.ActionStatusUnsigned
+	case "nosend":
+		return sdk.ActionStatusNoSend
+	case "nonfinal":
+		return sdk.ActionStatusNonFinal
+	default:
+		return sdk.ActionStatusUnprocessed
+	}
+}
+
+// mapActionInput maps wdk.WalletActionInput to sdk.ActionInput
+func mapActionInput(input wdk.WalletActionInput) sdk.ActionInput {
+	result := sdk.ActionInput{
+		SourceSatoshis:   input.SourceSatoshis,
+		InputDescription: input.InputDescription,
+		SequenceNumber:   input.SequenceNumber,
+	}
+
+	parts := strings.Split(input.SourceOutpoint, ":")
+	if len(parts) == 2 {
+		if txidHash, err := chainhash.NewHashFromHex(parts[0]); err == nil {
+			if vout, err := strconv.ParseUint(parts[1], 10, 32); err == nil {
+				result.SourceOutpoint = transaction.Outpoint{
+					Txid:  *txidHash,
+					Index: uint32(vout),
+				}
+			}
+		}
+	}
+
+	if input.SourceLockingScript != "" {
+		if lockingScript, err := script.NewFromHex(input.SourceLockingScript); err == nil {
```

please, wrap and return an error, instead of swallowing it